### PR TITLE
Update LINQ samples for collection expressions

### PR DESCRIPTION
**Pull request by BillWagner**
Contributes to #37000 

Mostly changed code, so the preview pages don't show updates. [Preview LINQ content](https://review.learn.microsoft.com/en-us/dotnet/csharp/linq/?branch=pr-en-us-37977)

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/programming-guide/concepts/linq/converting-data-types.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/converting-data-types.md) | [Converting Data Types (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/converting-data-types?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/filtering-data.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/filtering-data.md) | ["Filtering Data (C#)"](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/filtering-data?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/grouping-data.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/grouping-data.md) | [Grouping Data (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/grouping-data?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-combine-and-compare-string-collections-linq.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-combine-and-compare-string-collections-linq.md) | [How to combine and compare string collections (LINQ) (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-combine-and-compare-string-collections-linq?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-combine-linq-queries-with-regular-expressions.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-combine-linq-queries-with-regular-expressions.md) | [docs/csharp/programming-guide/concepts/linq/how-to-combine-linq-queries-with-regular-expressions](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-combine-linq-queries-with-regular-expressions?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-compare-the-contents-of-two-folders-linq.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-compare-the-contents-of-two-folders-linq.md) | [How to compare the contents of two folders (LINQ) (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-compare-the-contents-of-two-folders-linq?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-count-occurrences-of-a-word-in-a-string-linq.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-count-occurrences-of-a-word-in-a-string-linq.md) | ["How to count occurrences of a word in a string (LINQ) (C#)"](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-count-occurrences-of-a-word-in-a-string-linq?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-query-an-arraylist-with-linq.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-query-an-arraylist-with-linq.md) | [How to query an ArrayList with LINQ (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-query-an-arraylist-with-linq?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-query-for-duplicate-files-in-a-directory-tree-linq.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-query-for-duplicate-files-in-a-directory-tree-linq.md) | [docs/csharp/programming-guide/concepts/linq/how-to-query-for-duplicate-files-in-a-directory-tree-linq](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-query-for-duplicate-files-in-a-directory-tree-linq?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-query-for-files-with-a-specified-attribute-or-name.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-query-for-files-with-a-specified-attribute-or-name.md) | [How to query for files with a specified attribute or name (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-query-for-files-with-a-specified-attribute-or-name?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-query-for-sentences-that-contain-a-specified-set-of-words-linq.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-query-for-sentences-that-contain-a-specified-set-of-words-linq.md) | ["How to query for sentences that contain a specified set of words (LINQ) (C#)"](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-query-for-sentences-that-contain-a-specified-set-of-words-linq?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-query-for-the-total-number-of-bytes-in-a-set-of-folders-linq.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-query-for-the-total-number-of-bytes-in-a-set-of-folders-linq.md) | [How to query for the total number of bytes in a set of folders (LINQ) (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-query-for-the-total-number-of-bytes-in-a-set-of-folders-linq?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-query-the-contents-of-files-in-a-folder-lin.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-query-the-contents-of-files-in-a-folder-lin.md) | [docs/csharp/programming-guide/concepts/linq/how-to-query-the-contents-of-files-in-a-folder-lin](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-query-the-contents-of-files-in-a-folder-lin?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/how-to-sort-or-filter-text-data-by-any-word-or-field-linq.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/how-to-sort-or-filter-text-data-by-any-word-or-field-linq.md) | [How to sort or filter text data by any word or field (LINQ) (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-sort-or-filter-text-data-by-any-word-or-field-linq?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/projection-operations.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/projection-operations.md) | ["Projection operations (C#)"](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/projection-operations?branch=pr-en-us-37977) |
| [docs/csharp/programming-guide/concepts/linq/sorting-data.md](https://github.com/dotnet/docs/blob/08d773b8d0297518a1925aa44a907faec43f1b52/docs/csharp/programming-guide/concepts/linq/sorting-data.md) | [Sorting Data (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/sorting-data?branch=pr-en-us-37977) |

</details>

### CODE REVIEW

**Comment by BillWagner** [comment]
Ready for review, but waiting for G.A. to merge.

File: docs/csharp/programming-guide/concepts/linq/how-to-count-occurrences-of-a-word-in-a-string-linq.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ class CountWords
         string searchTerm = "data";  
   
         //Convert the string into an array of words  
-        string[] source = text.Split(new char[] { '.', '?', '!', ' ', ';', ':', ',' }, StringSplitOptions.RemoveEmptyEntries);  
+        string[] source = text.Split(['.', '?', '!', ' ', ';', ':', ','], StringSplitOptions.RemoveEmptyEntries);
```
Comment (IEvangelist) [inline docs/csharp/programming-guide/concepts/linq/how-to-count-occurrences-of-a-word-in-a-string-linq.md:33]:
```suggestion
        string[] source = text.Split(
            ['.', '?', '!', ' ', ';', ':', ','],
            StringSplitOptions.RemoveEmptyEntries);  
```

File: samples/snippets/csharp/VS_Snippets_VBCSharp/csLINQJoinOperation/CS/JoinOperation.cs [hunk 2]
```diff
@@ -1,129 +1,124 @@
...
-                };
+        public static void Example()
+        {
+            List<Product> products =
+            [
+                new Product { Name = "Cola", CategoryId = 0 },
+                new Product { Name = "Tea", CategoryId = 0 },
+                new Product { Name = "Apple", CategoryId = 1 },
+                new Product { Name = "Kiwi", CategoryId = 1 },
+                new Product { Name = "Carrot", CategoryId = 2 },
```
Comment (IEvangelist) [inline samples/snippets/csharp/VS_Snippets_VBCSharp/csLINQJoinOperation/CS/JoinOperation.cs:87]:
```suggestion
                new() { Name = "Cola", CategoryId = 0 },
                new() { Name = "Tea", CategoryId = 0 },
                new() { Name = "Apple", CategoryId = 1 },
                new() { Name = "Kiwi", CategoryId = 1 },
                new() { Name = "Carrot", CategoryId = 2 },
```

File: samples/snippets/csharp/VS_Snippets_VBCSharp/csLINQJoinOperation/CS/JoinOperation.cs [hunk 3]
```diff
@@ -1,129 +1,124 @@
...
-                {
-                    new Category { Id = 0, CategoryName = "Beverage" },
-                    new Category { Id = 1, CategoryName = "Fruit" },
-                    new Category { Id = 2, CategoryName = "Vegetable" }
-                };
+            List<Category> categories =
+            [
+                new Category { Id = 0, CategoryName = "Beverage" },
+                new Category { Id = 1, CategoryName = "Fruit" },
+                new Category { Id = 2, CategoryName = "Vegetable" }
```
Comment (IEvangelist) [inline samples/snippets/csharp/VS_Snippets_VBCSharp/csLINQJoinOperation/CS/JoinOperation.cs:94]:
```suggestion
                new() { Id = 0, CategoryName = "Beverage" },
                new() { Id = 1, CategoryName = "Fruit" },
                new() { Id = 2, CategoryName = "Vegetable" }
```

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**